test/k8sT/Services.go

```diff
@@ -152,6 +163,85 @@ var _ = Describe("K8sServicesTest", func() {
 				service.ExpectContains(clusterIP, "ClusterIP is not present in the cilium service list")
 			}
 		}, 300)
+
+		It("Checks connectivity is correct during deployment scale", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			type BackgroundTestAsserts struct {
+				res  *helpers.CmdRes
+				time time.Time
+			}
+			backgroundChecks := []*BackgroundTestAsserts{}
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				for {
+					select {
+					default:
+						res := kubectl.ExecPodCmd(
+							helpers.DefaultNamespace, appPods[helpers.App2],
+							helpers.CurlFail("http://%s/public", clusterIP))
+						assert := &BackgroundTestAsserts{
+							res:  res,
+							time: time.Now(),
+						}
+						backgroundChecks = append(backgroundChecks, assert)
```

Comment:
To prevent 100% CPU usage in case something fails and this enters an infinity loop, add a `time.Sleep(100*time.Miliseconds)` here

Reply:
Fixed